### ISSUE TITLE
[New Device Support] Request for No.1 S9

### ISSUE THREAD

**Issue by flotux2** [Freeyourgadget/Gadgetbridge#902]
Please support the No.1 model S9. It uses the MTK 2502 II and a MC3413-P as heart rate sensor.
The product web site is: http://en.001phone.cn/product-no-1-s9-nfc-smartwatch-131-html.

I would love to use Gadgetbridge as an alternative to the proprietary app Fundo Wear.

**Comment by protomors** [comment]
I am working on NO.1 F1 support for Gadgetbridge. If your device uses similar protocol it is possible to support both. Are you willing to help test it (build Gadgetbridge from source, provide logs)?

**Comment by flotux2** [comment]
Sounds promissing. :-)
I will receive my S9 form China in about 30 days. Then of course I will support you in testing the app. 
If I need to build something I obviously will need some initial help how to do it.

**Comment by ildar** [comment]
I hope I wont discourage you (too much).
MTK 2502/2503 is a very different platform. IMHO the workflow for a vendor usually looks like this:
> Choosing platform → Getting platform SDK+UX app SDK → Writing UIs and functions on top of SDKs → Sell,sell,sell!

Protocols usually encapsulated in SDK+UX SDK(aka Android/iOS app skeleton). Possibly the vendor can add something.
If my assumptions are correct, then:
1. protocols for different platforms might be very different
2. protocols for similar platforms may be very similar

Bottom line will be: No.1 F1 wouldn't help implementing no.1 S9

**Comment by ildar** [comment]
BTW, I have 2502 smart watch: G8.

**Comment by ildar** [comment]
Also for the protocol:
1. F1 is [DA14580](https://www.dialog-semiconductor.com/products/connectivity/bluetooth-low-energy/smartbond-da14580)
2. F4 is [NRF51822](https://www.nordicsemi.com/eng/Products/Bluetooth-low-energy/nRF51822)
3. S9/G8/many others are MT2502, G9 is coming with MT2503

**Comment by flotux2** [comment]
I finally received my S9 yesterday. If I can test or build anything let me know.

**Comment by ildar** [comment]
(update)
4. F5 is NRF52832QFAA

**Comment by protomors** [comment]
@ildar You are right. I looked at firmware files for S9 and they are structured almost like Android ROM for MTK smartphones :) Nothing like DA14580 firmware. So the protocol is probably also different. But I still can give it a try.

@flotux2 The easiest way to understand protocol is to look at how official app communicates with the device. Did you already try to use it? If so then you can try to record BT Snoop log during different actions (device pairing, data syncing) and send it to me. But keep in mind that logs can contain personal data. How to record logs is described in the wiki https://github.com/Freeyourgadget/Gadgetbridge/wiki/Log-Files in section "Android's Bluetooth Logging".

**Comment by ildar** [comment]
@protomors , just sent you my btsnoop privately.

**Comment by protomors** [comment]
@cpfeiffer This device uses virtual serial port (RFCOMM) and most things are done using AT commands. This means I need to use AbstractSerialDeviceSupport as base class. But for now I worked only with AbstractBTLEDeviceSupport. Which one of existing devices can I use as an example?

**Comment by ildar** [comment]
https://github.com/Freeyourgadget/Gadgetbridge/search?q=AbstractSerialDeviceSupport
says it's Pebble.

**Comment by flotux2** [comment]
@protomors, sorry for the late reply. Unfortunately I had to sell my No.1 S9 as the quality of the smart watch was below my expectations. 

Instead I ordered a Xiaomi Amazfit BIP that is actually somewhere between China and Europe. So hopefully I can contribute to this great project in another way.

Should I close this issue?

**Comment by ildar** [comment]
Don't. I take it over.

**Comment by vsmolovik** [comment]
> Also for the protocol:
> 
>     3. F5 is [NRF52832QFAA](https://www.nordicsemi.com/eng/Products/Bluetooth-low-energy/nRF52832)

Is there any chance to get supported watches with NRF52832? I can help in development is you can  give me directions...

Thx, Vlad.

**Comment by ildar** [comment]
Nrf smart bands already supported. E.g. I use custom build for DT58.
For Nordic family that use HPlus app please refer to
https://github.com/Freeyourgadget/Gadgetbridge/issues/1486
This issue should cover only MT2502/MT2503 family

**Comment by Fef0** [comment]
Hi guys, sorry for the necroposting but today I tried Gadgetbridge with my No. 1 model F4 and it didn't even find the watch. Am I doing something wrong (since you said that all the Nrf smart bands are already supported) or that specific model isn't supported? Thank you very much for your work, keep it up!

**Comment by ildar** [comment]
@Fef0, please move to
https://github.com/Freeyourgadget/Gadgetbridge/issues/1486
This issue should cover only MT2502/MT2503 family
